lms/services/http.py

```diff
@@ -0,0 +1,161 @@
+            form body.
+
+            Or it can be bytes or a file-like object (e.g. a JSON byte string)
+            it'll be used directly as the request body.
+
+        :param json: Data to send in the body of the request as JSON.
+
+            Using this changes the request's content-type to application/json.
+
+            This can't be used at the same time as the `data` argument.
+
+        :type json: Any JSON-serializable object (e.g. a list or dict)
+
+        :param headers: Headers to send in the request.
+
+            Header values should be byte strings not unicode.
+
+        :type headers: dict
+
+        :param auth: Authorization to send in the request's Authorization header.
+
+            This can be a (user, pass) 2-tuple which will be serialized using
+            HTTP Basic Authentication.
+
+            Or it can be a callable in order to implement other authentication
+            schemes. `requests` itself and libraries like `requests_oauthlib`
+            provide auth callables that you can use or you can write your own.
+
+            See: https://docs.python-requests.org/en/master/user/authentication/
+
+        :param timeout: How long (in seconds) to wait before raising an error.
+
+            This can be a (connect_timeout, read_timeout) 2-tuple or it can be
+            a single float that will be used as both the connect and read
+            timeout.
+
+            Note that the read_timeout is *not* a time limit on the entire
+            response download. It's a time limit on how long to wait *between
+            bytes from the server*. The entire download can take much longer.
+
+        :param schema: A schema class to use to validate the response
+        :type schema: lms.validation.RequestsResponseSchema
+
+        :param kwargs: Any other keyword arguments will be passed directly to
+            requests.Session().request():
+            https://docs.python-requests.org/en/latest/api/#requests.Session.request
+
+        :raise HTTPError: If sending the request or receiving the response
+            fails (DNS failure, refused connection, timeout, too many
+            redirects, etc).
+
+            The original exception from requests will be available as
+            HTTPError.__cause__.
+
+            In this case HTTPError.response will be None.
+
+        :raise HTTPError: If an error response (4xx or 5xx) is received.
+
+            The original exception from requests will be available as
+            HTTPError.__cause__.
+
+            The error response will be available as HTTPError.response.
+
+        :raise HTTPValidationError: If a `schema` argument is given and the
+            response fails validation with the schema.
+
+            The original lms.validation.ValidationError will be available as
+            HTTPValidationError.__cause__.
+
+            The invalid response will be available as
+            HTTPValidationError.response.
```

Comment:
Long docstring mostly because `requests.Session.request()` has a lot of arguments and I picked out the arguments that our code already uses or that I think we're likely to want to use and documented them here. I think this is useful because `requests`'s own documentation is kind of hard to navigate and you often have to look in multiple places in the `requests` docs to find all the pertinent information about one parameter. So this docstring acts as a kind of cheat sheet.

It also documents our own `schema` parameter and setting `validated_data`, and our custom exception-raising behaviour (the different types of exception raised and setting `__cause__` and `response`).